oxide_floating_ip: initial resource and data source (#427)
oxidecomputer/terraform-provider-oxide: docs/data-sources/oxide_floating_ip.md

---
page_title: "oxide_floating_ip Data Source - terraform-provider-oxide"
---

# oxide_floating_ip (Data Source)

Retrieve information about a specified floating IP.

## Example Usage

```hcl
data "oxide_floating_ip" "example" {
  project_name = "my-project"
  name         = "app-ingress"
}
```

## Schema

### Required

- `name` (String) Name of the floating IP.

> - `project_name` (String) Project name where this floating IP is located.

Reviewer: Nit: 

```suggestion
- `project_name` (String) Name of the project that contains the floating IP.
```

### Optional

- `timeouts` (Attribute, Optional) (see [below for nested schema](#nestedatt--timeouts))

### Read-Only

- `id` (String) Unique, immutable, system-controlled identifier for the floating IP.
- `description` (String) Description for the floating IP.
- `ip` (String) IP address for this floating IP. If unset an IP address will be chosen from the given `ip_pool_id`.
- `ip_pool_id` (String) IP pool ID to allocate this floating IP from. If unset the silo's default IP pool is used.

> - `instance_id` () Instance ID that this floating IP is attached to, if presently attached.

Reviewer: ```suggestion
- `instance_id` (String) Instance ID that this floating IP is attached to, if presently attached.
```
